frontend/src/components/Pages/MissionDefinitionPage/MissionDefinitionPage.tsx

```diff
@@ -48,21 +62,25 @@ const StyledMissionDefinitionPage = styled.div`
     gap: 1rem;
     margin: 2rem;
 `
+const StyledButton = styled(Button)`
+    width: 150px;
+`
+const StyledInspectionFrequencyDiv = styled.div`
+    > * {
+        padding: 10px;
+    }
+`
 
-interface IProps {
-    missionDefinition: CondensedMissionDefinition
-    updateMissionDefinition: (missionDefinition: CondensedMissionDefinition) => void
-}
-
-function KeyValuePairDisplay({ left, right }: { left: string; right: any }) {
+function KeyValuePairDisplay({ left, right, onEdit }: { left: string; right: any; onEdit?: () => void }) {
     return (
-        <>
+        <StyledFormItem>
             <Typography
                 variant="body_long_bold"
                 group="paragraph"
                 color={tokens.colors.text.static_icons__secondary.rgba}
             >
                 {left}
+                {onEdit && <Icon style={{ paddingLeft: '5px' }} name={Icons.Edit} size={16} onClick={onEdit}></Icon>}
```

Comment:
If the icon is clickable its better to make it a button with an icon such that the pointer indicates that its clickable